zebra-state/src/service/finalized_state/disk_format/block.rs

```diff
@@ -15,21 +15,46 @@ use zebra_chain::{
     transaction,
 };
 
-use crate::service::finalized_state::disk_format::{FromDisk, IntoDisk, IntoDiskFixedLen};
+use crate::service::finalized_state::disk_format::{
+    expand_zero_be_bytes, truncate_zero_be_bytes, FromDisk, IntoDisk, IntoDiskFixedLen,
+};
 
 #[cfg(any(test, feature = "proptest-impl"))]
 use proptest_derive::Arbitrary;
 
+/// The maximum value of an on-disk serialized [`Height`].
+///
+/// This allows us to store [`OutputIndex`]es in 8 bytes,
+/// which makes database searches more efficient.
+///
+/// # Consensus
+///
+/// This maximum height supports on-disk storage of blocks until around 2050.
+///
+/// Since Zebra only stores fully verified blocks on disk, blocks with heights
+/// larger than this height are rejected before reaching the database.
+/// (It would take decades to generate a valid chain this high.)
+pub const MAX_ON_DISK_HEIGHT: Height = Height(Height::MAX.0 / 256);
+
 // Transaction types
 
 /// A transaction's index in its block.
+///
+/// # Consensus
+///
+/// This maximum height supports on-disk storage of transactions in blocks up to ~5 MB.
```

Comment:
I assume this is based on the minimal size of a transaction.

Reply:
Specifically, the minimum size of a `Transaction::V1`. So the actual size limit for new blocks is a bit higher, because `V4` and `V5` transactions have more header fields.

Here's the minimum size calculation:
https://github.com/ZcashFoundation/zebra/blob/413f7fbb1dab7cfde07add0e9b39c870f7a9a566/zebra-chain/src/transaction/serialize.rs#L952-L957